### DISCUSSION TITLE
Beam with custom initial geometry

### DISCUSSION THREAD

**Discussion by Pawan2503** [InteractiveComputerGraphics/PositionBasedDynamics#165] [General]
Hi, I am interested to create beam with more complicated initial geometry, where the coordinates are avaialble in vector. Is this possible in pypbd? The beam example creates straight model only. Thank you for any advice

**Comment by janbender** [comment]
PositionBasedDynamics provides some functions (Utilities::TetGenLoader -> loadTetFile) to load a tet mesh. Such meshes can be generated from surface meshes using the tool TetGen (https://wias-berlin.de/software/index.jsp?id=TetGen&lang=1).
